libgoal/transactions.go

```diff
@@ -717,24 +717,14 @@ func (c *Client) MakeUnsignedAssetConfigTx(creator string, index uint64, newMana
 		params = asset.Params
 	} else {
 		// Fetch the current state, to fill in as a template
-		current, err := c.AccountInformationV2(creator, true)
+		current, err := c.AccountAssetInformation(creator, index)
```

Comment:
The change here is already an improvement, but I think the entire if/else statement can be removed and replaced with the contents of the current if branch (`c.AssetInformationV2`). Perhaps plus an additional assertion that if `creator` is not empty, it matches the asset creator.

Reply:
Good point - I took this suggestion in [8e97a9f](https://github.com/algorand/go-algorand/pull/4758/commits/8e97a9f927c54809cf8fe91ef98896ddf0fff7d5)